### PR TITLE
Added CLI flags for updating multiple domains at once.

### PR DESCRIPTION
**Pull request by jdevelop**

### CODE REVIEW

File: dyndns53.go [hunk 1]
```diff
@@ -17,37 +17,65 @@ import (
...
+	return strings.Join([]string(*i), ",")
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
 func main() {
+
```
Comment (agorf) [inline dyndns53.go]: Could you remove this blank line?

File: dyndns53.go [hunk 2]
```diff
@@ -17,37 +17,65 @@ import (
...
+	progName          = "dyndns53"
+	ipFileNameDefault = "." + progName + "-ip"
 )
 
+var (
+	recSet     recordSet
+	logFn      string
+	awsProfile string
+	ipFileName string
+	names      arrayFlags
```
Comment (agorf) [inline dyndns53.go]: `names` can be moved inside `main` and not be a global.

File: dyndns53.go [hunk 3]
```diff
@@ -17,37 +17,65 @@ import (
...
+	*i = append(*i, value)
+	return nil
+}
+
 func main() {
+
 	log.SetPrefix(progName + ": ")
 	log.SetFlags(0)
 
-	var recSet recordSet
```
Comment (agorf) [inline dyndns53.go]: It's better to leave this here since the other methods like `upsert` and `validate` are methods of `recSet`, so they have access to it.

File: dyndns53.go (hunk 2 above)
Comment (agorf) [reply dyndns53.go]: It's better to remove `recSet`. See my other comment.

File: dyndns53.go [hunk 4]
```diff
@@ -17,37 +17,65 @@ import (
...
-	var logFn string
-	flag.StringVar(&recSet.name, "name", "", "record set name (domain)")
+	flag.Var(&names, "name", "record set names (-name domain1 -name domain2 -name domain3 ...)")
 	flag.StringVar(&recSet.rsType, "type", "A", `record set type; "A" or "AAAA"`)
 	flag.Int64Var(&recSet.ttl, "ttl", 300, "TTL (time to live) in seconds")
 	flag.StringVar(&recSet.hostedZoneID, "zone", "", "hosted zone id")
 	flag.StringVar(&logFn, "log", "", "file name to log to (default is stdout)")
+	flag.StringVar(&awsProfile, "profile", progName, "AWS profile")
```
Comment (agorf) [inline dyndns53.go]: It's better to break this and its related changes in `upsert` on a separate commit that adds support for a custom profile. It's not really related with updating many domains at once.

File: dyndns53.go [hunk 5]
```diff
@@ -17,37 +17,65 @@ import (
...
+var (
+	recSet     recordSet
+	logFn      string
+	awsProfile string
+	ipFileName string
+	names      arrayFlags
+)
+
+func (i *arrayFlags) String() string {
+	return strings.Join([]string(*i), ",")
```
Comment (agorf) [inline dyndns53.go]: Better to join them with a space ` ` or `, ` (space after comma).

File: dyndns53.go [hunk 6]
```diff
@@ -17,37 +17,65 @@ import (
...
-	var recSet recordSet
-	var logFn string
-	flag.StringVar(&recSet.name, "name", "", "record set name (domain)")
+	flag.Var(&names, "name", "record set names (-name domain1 -name domain2 -name domain3 ...)")
 	flag.StringVar(&recSet.rsType, "type", "A", `record set type; "A" or "AAAA"`)
 	flag.Int64Var(&recSet.ttl, "ttl", 300, "TTL (time to live) in seconds")
 	flag.StringVar(&recSet.hostedZoneID, "zone", "", "hosted zone id")
 	flag.StringVar(&logFn, "log", "", "file name to log to (default is stdout)")
+	flag.StringVar(&awsProfile, "profile", progName, "AWS profile")
+	flag.StringVar(&ipFileName, "ip-file", ipFileNameDefault, "IP file storage")
```
Comment (agorf) [inline dyndns53.go]: It's better to break this and its related changes in `upsert` on a separate commit that adds support for a custom ip file. It's not really related with updating many domains at once.

Comment (agorf) [reply dyndns53.go]: Can you document these 3 changes to the flags to the README file?

File: dyndns53.go [hunk 7]
```diff
@@ -114,37 +142,39 @@ func updateLastIPAddress(ip string) error {
...
-						Name: aws.String(rs.name),
-						Type: aws.String(rs.rsType),
-						TTL:  aws.Int64(rs.ttl),
-						ResourceRecords: []*route53.ResourceRecord{
-							{
-								Value: aws.String(rs.value),
-							},
-						},
+	changes := make([]*route53.Change, len(recSet.names))
+	for i, v := range recSet.names {
```
Comment (agorf) [inline dyndns53.go]: Better to name `v` as `name` for readability.

File: dyndns53.go [hunk 8]
```diff
@@ -156,11 +186,13 @@ func (rs *recordSet) upsert() (*route53.ChangeResourceRecordSetsOutput, error) {
...
 
 func (rs *recordSet) validate() error {
-	if rs.name == "" {
-		return fmt.Errorf("missing record set name")
-	}
-	if !strings.HasSuffix(rs.name, ".") {
-		return fmt.Errorf(`record set name must end with a "."`)
+	for _, v := range rs.names {
+		if v == "" {
+			return fmt.Errorf("missing record set name")
```
Comment (agorf) [inline dyndns53.go]: It's better to provide an index here since now there are many names. Something like:
```suggestion
			return fmt.Errorf("missing record set name at index %d", i)
```
Don't forget to change `_` to `i`.

File: dyndns53.go [hunk 9]
```diff
@@ -156,11 +186,13 @@ func (rs *recordSet) upsert() (*route53.ChangeResourceRecordSetsOutput, error) {
...
+		}
+		if !strings.HasSuffix(v, ".") {
+			return fmt.Errorf(`record set name must end with a "."`)
```
Comment (agorf) [inline dyndns53.go]: I think this is not necessary since inside `main` we strip `.` and append it again. Ideally this should be on a separate commit before the commit that adds support for many names.

File: dyndns53.go [hunk 10]
```diff
@@ -17,37 +17,65 @@ import (
...
 	if len(os.Args) == 1 {
 		flag.Usage()
 		os.Exit(1)
 	}
 	flag.Parse()
 
-	recSet.name = strings.TrimSuffix(recSet.name, ".") + "." // append . if missing
+	recSet.names = make([]string, len(names))
+
+	for i, v := range names {
```
Comment (agorf) [inline dyndns53.go]: It's better to name `v` as `name` for readability.

File: dyndns53.go [hunk 11]
```diff
@@ -114,37 +142,39 @@ func updateLastIPAddress(ip string) error {
 	return nil
 }
 
-func (rs *recordSet) upsert() (*route53.ChangeResourceRecordSetsOutput, error) {
+func (rs *recordSet) upsert(profileName string) (*route53.ChangeResourceRecordSetsOutput, error) {
```
Comment (agorf) [inline dyndns53.go]: It's better to be consistent and name this `awsProfile` as well.

File: dyndns53.go (hunk 3 above)
Comment (jdevelop) [reply dyndns53.go]: Done, moved into the variable block below

File: dyndns53.go (hunk 1 above)
Comment (jdevelop) [reply dyndns53.go]: Done

File: dyndns53.go (hunk 4 above)
Comment (jdevelop) [reply dyndns53.go]: Done

File: dyndns53.go (hunk 5 above)
Comment (jdevelop) [reply dyndns53.go]: Done

File: dyndns53.go (hunk 7 above)
Comment (jdevelop) [reply dyndns53.go]: Done

File: dyndns53.go (hunk 8 above)
Comment (jdevelop) [reply dyndns53.go]: Done

File: dyndns53.go (hunk 9 above)
Comment (jdevelop) [reply dyndns53.go]: Removed

File: dyndns53.go (hunk 10 above)
Comment (jdevelop) [reply dyndns53.go]: Done

File: dyndns53.go (hunk 11 above)
Comment (jdevelop) [reply dyndns53.go]: Removed

**Comment by agorf** [comment]
@jdevelop I've merged your changes to master, keeping only the support for many domains. Feel free to issue another pull request for specifying a custom AWS profile path and a custom log file path (please put them on two separate commits). Thanks!
